[Design] 대학교 리스트 화면 UI

tea-hkim opened the pull request:
## 작업내용
<!-- 작업 내용과 이미지를 첨부해주세요. -->
- [x] UI 작업만 진행된 상태
- [ ] UseCase와 모델 작업은 추후에 진행할 예정입니다

<!-- (+스크린샷)이 있다면 적어주세요. 없으면 지워주세요-->
|작업 전|작업 후|
|:---:|:---:|
|<img width="250" src="https://user-images.githubusercontent.com/81206228/196221460-c61a1474-0e8b-4c93-8841-e754b3307e79.png">|<img width="250" src="https://user-images.githubusercontent.com/81206228/199876873-b31677ac-6dbe-40bf-b34c-94f83f18d394.png">|
|iPhone 13 pro max| iPhone SE 3|
|<img width="250" src="https://user-images.githubusercontent.com/81206228/199876814-20bf890b-7f24-4b06-bbda-91cae6830f59.png">|<img width="250" src="https://user-images.githubusercontent.com/81206228/199876945-14d1227b-73e4-4752-8f00-0b471e96adb2.png">|

## 리뷰포인트
<!-- 리뷰가 필요한 포인트와 해당 되는 커밋을 링크로 걸어주세요. -->
### UIView extension empty View 레이아웃 변경
```
(
    "<NSAutoresizingMaskLayoutConstraint:0x60000023a3f0 h=--& v=--& UIView:0x139a09320.width == 0   (active)>",
    "<NSLayoutConstraint:0x600000234640 H:|-(20)-[UILabel:0x139813160]   (active, names: '|':UIView:0x139a09320 )>",
    "<NSLayoutConstraint:0x600000234690 UILabel:0x139813160.trailing == UIView:0x139a09320.trailing - 20   (active)>"
)
```
위와 같은 레이아웃 경고가 발생해서 emptyView를 확인해보니 너비가 설정되어 있지 않은데 내부 컴포넌트들에 레이아웃이 설정되어있어 발생한 에러였습니다.
emptyView를 화면 중앙에 위치시키고 width를 200point로 설정하여 레이아웃 경고를 해결했습니다

## Reference
<!-- 참고한 자료를 작성해주세요 -->

## Checklist
- [x] 브랜치를 가져와 작업한 경우 이전 브랜치에 PR을 보냈는지 확인
- [x] 빌드를 위해 SceneDelegate 수정한 것 PR로 올리지 않았는지 확인
- [x] 필요없는 주석, 프린트문 제거했는지 확인
- [x] 컨벤션 지켰는지 확인
- [x] final, private 제대로 넣었는지 확인
- [x] 다양한 디바이스에 레이아웃이 대응되는지 확인
  - [x] iPhone SE
  - [x] iPhone 13
  - [x] iPhone 13 Pro Max

tea-hkim committed:
Feat:sparkles:: navigationBarTitle 변경

네비게이션바 타이틀을 변경하였습니다
  - 타이틀을 대학 선택으로 변경하였습니다
  - BackButton의 타이틀을 수정하였습니다
  - BackButton의 tintColor를 검은색으로 변경하였습니다

tea-hkim committed:
Feat:sparkles:: navigationBar를 수정하는 함수 추가

네비게이션바 UI를 변경하는 함수를 추가하였습니다
  - 따로 함수를 만들었습니다
  - 네비게이션바 타이틀의 속성을 변경하였습니다

tea-hkim committed:
Feat:sparkles:: SearchTextField 추가

SearchTextField을 추가하였습니다
  - SearchButton을 추가하였습니다

tea-hkim committed:
Feat:sparkles:: SearchTextField 수정

  - SearchTextField의 placeholder가 변경이 되지 않아 수정을 하였습니다

tea-hkim committed:
Feat:sparkles:: OrganizationListCell에 combine 추가

 - cancelBag을 추가하였습니다
 - prepareForReuse 함수를 추가해줬습니다
 - UI Function 마크의 위치를 함수 밖으로 변경했습니다

tea-hkim committed:
Feat:sparkles:: 대학교 검색 기본 이미지뷰, 검색 결과 없음 뷰 추가

tea-hkim committed:
Chore:broom:: 의미없는 줄바꿈 삭제

tea-hkim committed:
Fix:beetle:: UIView+ emptyView 레이아웃 깨지는 현상 해결

 - emptyView의 width가 정해져 있지 않고 내부의 component들이 레이아웃이 지정이 되어있어 레이아웃이 깨지는 현상이 있습니다.
 - emptyView를 화면 중앙에 위치시키고 width를 200point로 설정하여 레이아웃 경고를 해결했습니다

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/ViewModel/OrganizatinoListViewModel.swift:
```diff
@@ -24,7 +24,7 @@ final class OrganizationListViewModel {
     
     init() {
```
Q
이 뷰 모델에 있는 주석들은 추후 연결작업할 떄 필요한거죠?

tea-hkim replied:
넵 그렇습니다! 아직 뷰와 모델을 연결하지 않아서 간단하게 연결해놨습니다

Gobans commented on Projects/App/Sources/UI/Organization/OrganizationList/View/OrganizationListViewController.swift:
```diff
@@ -94,33 +103,52 @@ extension OrganizationListViewController {
+        navigationItem.title = "대학 선택"
+        self.navigationController?.navigationBar.titleTextAttributes = [
+            NSAttributedString.Key.font: UIFont.systemFont(ofSize: 17, weight: .medium)
+        ]
```
p2
이거 global set 같은데 가능하다면 (모든 뷰의 네비게이션 타이틀 특성이 동일하다면) 앱이 실행되는 진입뷰에서 쓰이는게 좋을 것 같아요
아니면 현재 클래스를 특정해서 attribute를 set을 해주면 좋을 것 같아요

tea-hkim replied:
변경했습니다! a3434815aff44d8449ec4d646125179197135e34
말씀해주신대로 네비게이션 뒤로가기 버튼은 다른 뷰에서도 전부 쓰이는 코드라서 SceneDelegate로 옮겨서 처리했습니다
타이틀을 설정하는 코드는 다른 뷰에서는 공통으로 사용되는 부분이 적어서 configureUI로 옮겼습니다

Gobans reviewed (commented):
empty View 레이아웃 고친줄 알았는데 아니였군요..!
고쳐주셔서 감사합니다

tea-hkim committed:
Feat:sparkles:: navigationbar back button 설정 변경

 - 공통으로 사용되는 네비게이션바 뒤로가기 버튼의 설정을 SceneDelegate로 옮겼습니다.
 - OrganizationListViewController에 있던 setNav 함수를 삭제하고 네비게이션 타이틀을 설정하는 코드를 configureUI로 옮겼습니다.

Chaeho-Min reviewed (approved):
이 검색창은 세 번째 보는거 같네요 ㅋㅋㅋㅋ 수고하셨습니다~
